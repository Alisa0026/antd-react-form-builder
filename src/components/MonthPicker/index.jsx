import React from 'react'
import _ from 'lodash'
import moment from 'moment'
import BasicItem from '../BasicItem'
import AntdDatePicker from 'antd/lib/date-picker'
import 'antd/lib/date-picker/style/css'
import FormItemComponentDecorator from '../../decorator/FormItemComponent'

const { MonthPicker, RangePicker } = AntdDatePicker;

function component(BasicItemComponent){
  @FormItemComponentDecorator
  class FMonthPicker extends React.Component {

    render(){
      let { ...other } = this.props;
      other.targetComponent = MonthPicker;
      other.type = "monthpicker";
      if(_.isString(other.value)){
        other.value = new moment(other.value,other.format || "YYYY-MM")
      }
      this.propsAdapter(other);
      return (
        <BasicItemComponent { ...other }/>
      )
    }
    
  }
  return FMonthPicker;
}

export default component(BasicItem) 




