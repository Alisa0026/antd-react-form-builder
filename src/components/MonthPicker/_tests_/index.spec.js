import React from 'react'
import { shallow,mount,render } from 'enzyme'
import sinon from 'sinon'
import moment from 'moment'
import LocaleProvider from 'antd/lib/locale-provider'
import { 
  testCommonProps, 
  getAntdFormComponentMount,
  getAntdFormItemMount,
} from '../../../../tests/common'
import MonthPicker from '../index'

describe('MonthPicker',function(){
  sinon.stub(console,"warn");
  it('renders correctly', () => {
    const wrapper = render(
      <MonthPicker 
        required
        className="test"
        label="name"
        name="test"
        value="2016-04"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  testCommonProps(MonthPicker); 
  it("stirng value should be rendered correctly",function(){
    const wrapper = mount(
      <MonthPicker 
        required
        className="test"
        label="name"
        name="test"
        value="2016-04"
      />
    );
    const antdInputMount = getAntdFormComponentMount(wrapper,1);
    const antdFormItemMount = getAntdFormItemMount(wrapper);
    //console.log(antdInputMount.pr)
    expect(antdInputMount.props().value).toEqual(new moment("2016-04","YYYY-MM"));
  })
  it("moment value should be rendered correctly",function(){
    const wrapper = mount(
      <MonthPicker 
        required
        className="test"
        label="name"
        name="test"
        value={ moment("2016-04") }
      />
    );
    const antdInputMount = getAntdFormComponentMount(wrapper,1);
    const antdFormItemMount = getAntdFormItemMount(wrapper);
    //console.log(antdInputMount.pr)
    expect(antdInputMount.props().value).toEqual(moment("2016-04"));
  })
  it("should have hidden <Input /> ",function(){
    var value = new moment("2016-04");
    const wrapper = mount(
      <MonthPicker 
        required
        className="test"
        label="name"
        name="test"
        value={ value }
      />
    );
    const antdInputMount = getAntdFormComponentMount(wrapper,1);
    const antdFormItemMount = getAntdFormItemMount(wrapper);
    //console.log(antdInputMount.pr)
    expect(wrapper.childAt(1).props().value).toEqual(Math.floor(+value / 1000));
    expect(wrapper.childAt(1).props().name).toEqual('test');
  })
})

