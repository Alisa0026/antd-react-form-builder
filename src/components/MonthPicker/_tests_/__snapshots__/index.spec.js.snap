// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthPicker renders correctly 1`] = `
<span>
  <div
    class="ant-row ant-form-item"
  >
    <div
      class="ant-form-item-label"
    >
      <label
        class="ant-form-item-required"
        title="name"
      >
        name
      </label>
    </div>
    <div
      class="ant-form-item-control-wrapper"
    >
      <div
        class="ant-form-item-control "
      >
        <span
          class="ant-calendar-picker"
        >
          <span>
            <input
              class="ant-calendar-picker-input ant-input"
              placeholder="请选择日期"
              readonly=""
              value=""
            />
            <span
              class="ant-calendar-picker-icon"
            />
          </span>
        </span>
      </div>
    </div>
  </div>
  <input
    class="ant-input"
    name="test"
    type="hidden"
    value="NaN"
  />
</span>
`;
